### PR TITLE
[130] Allow user to create a photo

### PR DESCRIPTION
**Pull request by salcudeann**
## Description
Fix issues that can be found in [this PR](https://github.com/Agilefreaks/FreaksCatalog-API/pull/23). 
Add a mutation to create a photo with next fields: `imageableId`, `imageableType`, `uri`. `imageableType` represents the type of object where is assigned this photo and the `imageableId` represents the id of the object.

## Related

[[42] Add photo to a freak](https://trello.com/c/0yKfo1ER/130-add-photo-to-a-freak)

## Steps to Test or Reproduce
Create a photo entry with next GraphQl mutation:
```GraphQl
mutation {
  photoCreate(imageableId: 1, imageableType:"Freak", uri: "https://www.gravatar.com/avatar/00000000000000000000000000000000") {
    imageableType
    imageableId
    uri
  }
}

```

### CODE REVIEW

**Comment by gabrielgiroe1** [comment]
Please resolve warnings from guard

**Comment by gabrielgiroe1** [comment]
Please put Guardfile in gitignore

**Review by gabrielgiroe1 — APPROVED** [review]

File: app/graphql/types/freak_type.rb [hunk 1]
```diff
@@ -8,7 +8,7 @@ class FreakType < Types::BaseObject
     field :last_name, String, null: false
     field :description, String, null: false
     field :email, String, null: false
-    field :photo, PhotoType, null: false
+    field :photo, PhotoType, null: true
```
Comment (pirvudoru) [inline app/graphql/types/freak_type.rb:11]: This is considered a breaking change :)

File: app/graphql/types/url.rb [hunk 2]
```diff
@@ -6,7 +6,8 @@ class Url < Types::BaseScalar
 
     def self.coerce_input(input_value, _context)
       url = URI.parse(input_value)
-      return if url.is_a?(URI::HTTP) || url.is_a?(URI::HTTPS)
+
+      return url if url.is_a?(URI::HTTP) || url.is_a?(URI::HTTPS)
```
Comment (pirvudoru) [inline app/graphql/types/url.rb:10]: You don't need `return`

File: spec/controllers/graphql/mutations/photo_create_spec.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  RSpec.describe GraphqlController, type: :controller do
+    subject { post :execute, params: params, as: :json }
+
+    let(:params) do
+      {
+        query: File.read('spec/fixtures/requests/mutations/photo_create.graphql')
+      }
+    end
+
+    it { is_expected.to match_response_for(mutation: :photo_create, sample: :photoCreate) }
```
Comment (pirvudoru) [inline spec/controllers/graphql/mutations/photo_create_spec.rb:15]: please use snake_case

File: spec/fixtures/requests/mutations/photo_create.graphql [hunk 4]
```diff
@@ -0,0 +1,7 @@
+mutation {
+    photoCreate(imageableId: 1, imageableType:"Freak", uri: "https://www.gravatar.com/avatar/00000000000000000000000000000000") {
```
Comment (pirvudoru) [inline spec/fixtures/requests/mutations/photo_create.graphql:2]: So we don't care if the imageableType+imageableId actually exist in the database.

File: app/graphql/types/photo_type.rb [hunk 5]
```diff
@@ -3,6 +3,8 @@
 module Types
   class PhotoType < Types::BaseObject
     field :id, GraphQL::Types::ID, null: false
+    field :imageable_type, String, null: false
+    field :imageable_id, String, null: false
```
Comment (alex-damian-negru) [inline app/graphql/types/photo_type.rb:7]: This should be of the same type as the mutation argument
```ruby
argument :imageable_id, GraphQL::Types::ID, required: true
```

Comment (alex-damian-negru) [reply app/graphql/types/photo_type.rb:7]: Also, how can one know the possible values for `imageable_type`? :)

**Review by alex-damian-negru — CHANGES_REQUESTED** [review]

**Comment by alex-damian-negru** [comment]
[Add Photo to a Freak](https://trello.com/c/0yKfo1ER/130-add-photo-to-a-freak)

File: app/graphql/resolvers/photo_create_resolver.rb [hunk 6]
```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+module Resolvers
+  class PhotoCreateResolver < Resolvers::Base
+    argument :imageable_id, Integer, required: true
+    argument :imageable_type, String, required: true
```
[the remark this answers is not in the document]
Comment (alex-damian-negru) [reply app/graphql/resolvers/photo_create_resolver.rb:7]: This is not a descripton. Look over https://graphql-ruby.org/fields/introduction.html#field-documentation

**Review by alex-damian-negru — CHANGES_REQUESTED** [review]

File: app/graphql/resolvers/photo_create_resolver.rb (hunk 6 above)
[the remark this answers is not in the document]
Comment (salcudeann) [reply app/graphql/resolvers/photo_create_resolver.rb]: Done :+1:

File: app/graphql/resolvers/photo_create_resolver.rb [hunk 7]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module Resolvers
+  class PhotoCreateResolver < Resolvers::Base
+    argument :imageable_id, GraphQL::Types::ID, required: true
+    argument :imageable_type, String, 'Can have value Freak or Project, depending where the photo is assigned',
+             required: true
+    argument :uri, Types::Url, required: true
```
Comment (pirvudoru) [inline app/graphql/resolvers/photo_create_resolver.rb:8]: We will need to add validations for these arguments. Currently someone can add anything in the fields. Ticket for another time.

**Pull request merged**